src/pages/index.js

```diff
@@ -38,75 +51,159 @@ const addCardFormValidator = new FormValidator(
 );
 addCardFormValidator.enableValidation();
 
-// Initialize PopupWithForm for editing profile
+const avatarFormValidator = new FormValidator(validationSettings, avatarForm);
+avatarFormValidator.enableValidation();
+
+// Popups
 const profileEditPopup = new PopupWithForm(
   "#profile-edit-popup",
   handleProfileEditSubmit,
   profileEditFormValidator
 );
 profileEditPopup.setEventListeners();
 
-// Initialize PopupWithForm for adding new cards
 const addCardPopup = new PopupWithForm(
   "#profile-add-popup",
   handleAddCardFormSubmit,
   addCardFormValidator
 );
 addCardPopup.setEventListeners();
 
-// Initialize PopupWithImage for previewing images
 const imagePopup = new PopupWithImage(".js-preview-popup");
 imagePopup.setEventListeners();
 
-// Function to handle image clicks
+const avatarPopup = new PopupWithForm(
+  "#edit-avatar-popup",
+  handleAvatarFormSubmit,
+  avatarFormValidator
+);
+avatarPopup.setEventListeners();
+
+// Functions
 function handleImageClick(cardData) {
   imagePopup.open(cardData);
 }
 
-// Function to create a new card
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    { ...cardData, currentUserId: userInfo.getUserInfo()._id },
+    "#card-template",
+    handleImageClick,
+    {
+      handleLikeClick: (cardId) => {
+        const likePromise = cardData.isLiked
+          ? api.dislikeCard(cardId)
+          : api.likeCard(cardId);
+        likePromise
+          .then((updatedCard) => {
+            card.updateLikes(updatedCard.isLiked);
+          })
+          .catch((err) => {
+            console.error("Error toggling like:", err);
+          });
+      },
+      handleDeleteClick: (cardId) => {
+        deleteCardPopup.open();
+        deleteCardPopup.setSubmitAction(() => {
+          api
+            .deleteCard(cardId)
+            .then(() => {
+              card.removeCard();
+              deleteCardPopup.close();
+            })
+            .catch((err) => console.error("Error deleting card:", err));
+        });
+      },
+    }
+  );
   return card.generateCard();
 }
 
-// Section instance to render cards
-const section = new Section(
-  { items: initialCards, renderer: createCard },
-  ".cards__list"
-);
-
-// Handle profile edit form submission
 function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo({
-    name: data.title || "",
-    job: data.description || "",
-  });
+  profileEditPopup.setLoadingState(true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```